provider/cmd/pulumi-gen-aws-native/main.go

```diff
@@ -96,69 +97,75 @@ func main() {
 		if err != nil {
-			panic(fmt.Errorf("error reading CloudFormation docs file: %v", err))
+			fatalf("error reading CloudFormation docs file: %v", err)
 		}
 		semanticsDocument, err := schema.GatherSemantics(semanticsDir)
 		if err != nil {
-			panic(fmt.Errorf("error gathering semantics: %v", err))
+			fatalf("error gathering semantics: %v", err)
 		}
 
 		packageSpec, meta, reports, err := schema.GatherPackage(supportedTypes, jsonSchemas, false, &semanticsDocument, docsTypes)
 		if err != nil {
-			panic(fmt.Errorf("error generating schema: %v", err))
+			fatalf("error generating schema: %v", err)
 		}
 
 		fmt.Println("Generating examples...")
 		err = generateExamples(packageSpec, meta, []string{"nodejs", "python", "dotnet", "go"})
 		if err != nil {
-			panic(fmt.Errorf("error generating examples: %v", err))
+			fatalf("error generating examples: %v", err)
 		}
 		providerDir := filepath.Join(".", "provider", "cmd", "pulumi-resource-aws-native")
 		if err = writePulumiSchema(*packageSpec, providerDir, true /* includeUncompressed */); err != nil {
-			panic(fmt.Errorf("error writing schema: %v", err))
+			fatalf("error writing schema: %v", err)
 		}
 
 		cf2pulumiDir := filepath.Join(".", "provider", "cmd", "cf2pulumi")
 		if err = writePulumiSchema(*packageSpec, cf2pulumiDir, false /* includeUncompressed */); err != nil {
-			panic(fmt.Errorf("error writing schema: %v", err))
+			fatalf("error writing schema: %v", err)
 		}
 		if err = writePulumiSchema(*packageSpec, "bin", false /* includeUncompressed */); err != nil {
-			panic(fmt.Errorf("error writing schema: %v", err))
+			fatalf("error writing schema: %v", err)
 		}
 
 		// Emit the resource metadata for cf2pulumi.
 		if err = writeMetadata(meta, cf2pulumiDir, false /* includeUncompressed */); err != nil {
-			panic(fmt.Errorf("error writing metadata: %v", err))
+			fatalf("error writing metadata: %v", err)
 		}
 		// Also, emit the resource metadata for the provider.
 		if err = writeMetadata(meta, providerDir, true /* includeUncompressed */); err != nil {
-			panic(fmt.Errorf("error writing metadata: %v", err))
+			fatalf("error writing metadata: %v", err)
 		}
 		// Also write to bin folder to be picked up as an asset for testing.
 		if err = writeMetadata(meta, "bin", true /* includeUncompressed */); err != nil {
-			panic(fmt.Errorf("error writing metadata: %v", err))
+			fatalf("error writing metadata: %v", err)
 		}
 		// Emit the reports for the provider.
 		if err := reports.WriteToDirectory("reports"); err != nil {
-			panic(fmt.Errorf("error writing reports: %v", err))
+			fatalf("error writing reports: %v", err)
 		}
 
 	default:
-		panic(fmt.Sprintf("Unrecognized language '%s'", operation))
+		fatalf("Unrecognized language '%s'", operation)
 	}
 }
 
+// Fail loud and clear.
+func fatalf(format string, a ...any) {
+	barrier := strings.Repeat("=", 110)
+	log.Fatalf(fmt.Sprintf("schema generation failed\n%s\n%s\n%s\n", barrier, fmt.Sprintf(format, a...), barrier))
+}
```

Comment:
Are we not concerned that jobs that used to depend on hard failure from this step will now pass with a silent failure in the logs?

Reply:
log.Fatalf is similar to panic in setting exit code 1.

Reply:
I think it just adds a date. We could switch back to panic if that's better (it has stack trace). But these errors I think seem to be sufficiently annotated with wrappers that stack trace is not needed.